packages/blade/src/components/Box/BaseBox/getBaseBoxStyles.tsx

```diff
@@ -0,0 +1,211 @@
+    gridRowEnd: getResponsiveValue(props.gridRowEnd, size),
+    gridArea: getResponsiveValue(props.gridArea, size),
+    gridAutoFlow: getResponsiveValue(props.gridAutoFlow, size),
+    gridAutoRows: getResponsiveValue(props.gridAutoRows, size),
+    gridAutoColumns: getResponsiveValue(props.gridAutoColumns, size),
+    gridTemplate: getResponsiveValue(props.gridTemplate, size),
+    gridTemplateAreas: getResponsiveValue(props.gridTemplateAreas, size),
+    gridTemplateColumns: getResponsiveValue(props.gridTemplateColumns, size),
+    gridTemplateRows: getResponsiveValue(props.gridTemplateRows, size),
+
+    // Spacing Props
+    padding: getSpacingValue(props.padding, props.theme, size),
+    paddingTop: getSpacingValue(props.paddingTop ?? props.paddingY, props.theme, size),
+    paddingBottom: getSpacingValue(props.paddingBottom ?? props.paddingY, props.theme, size),
+    paddingRight: getSpacingValue(props.paddingRight ?? props.paddingX, props.theme, size),
+    paddingLeft: getSpacingValue(props.paddingLeft ?? props.paddingX, props.theme, size),
+    margin: getSpacingValue(props.margin, props.theme, size),
+    marginBottom: getSpacingValue(props.marginBottom, props.theme, size),
+    marginTop: getSpacingValue(props.marginTop, props.theme, size),
+    marginRight: getSpacingValue(props.marginRight, props.theme, size),
+    marginLeft: getSpacingValue(props.marginLeft, props.theme, size),
+    height: getSpacingValue(props.height, props.theme, size),
+    minHeight: getSpacingValue(props.minHeight, props.theme, size),
+    maxHeight: getSpacingValue(props.maxHeight, props.theme, size),
+    width: getSpacingValue(props.width, props.theme, size),
+    minWidth: getSpacingValue(props.minWidth, props.theme, size),
+    maxWidth: getSpacingValue(props.maxWidth, props.theme, size),
+    gap: getSpacingValue(props.gap, props.theme, size),
+    rowGap: getSpacingValue(props.rowGap, props.theme, size),
+    columnGap: getSpacingValue(props.columnGap, props.theme, size),
+    top: getSpacingValue(props.top, props.theme, size),
+    right: getSpacingValue(props.right, props.theme, size),
+    bottom: getSpacingValue(props.bottom, props.theme, size),
+    left: getSpacingValue(props.left, props.theme, size),
+
+    // Visual props
+    backgroundColor: getBackgroundValue(props.backgroundColor, props.theme, size),
+    borderRadius: props.borderRadius
+      ? makeSize(
+          getIn(props.theme, `border.radius.${getResponsiveValue(props.borderRadius, size)}`),
+        )
+      : undefined,
+    transform: getResponsiveValue(props.transform, size),
+  };
+};
+
+const getAllMediaQueries = (props: BaseBoxProps & { theme: Theme }): CSSObject => {
+  if (isReactNative()) {
+    return {};
+  }
+
+  const { base, ...breakpointsWithoutBase } = breakpoints;
+
+  return Object.fromEntries(
+    Object.entries(breakpointsWithoutBase).map(([breakpointKey, breakpointValue]) => {
+      const mediaQuery = `@media ${getMediaQuery({ min: breakpointValue })}`;
```

Comment:
Thinking if we should keep `@media` outside of the interpolation in `getMediaQuery` or if we should also add that in `getMediaQuery`?

Reply:
we have to because of this - https://github.com/razorpay/blade/pull/1009#discussion_r1111752514